### DISCUSSION TITLE
Setting language-server-specific settings

### DISCUSSION THREAD

**Discussion by aidalgol** [joaotavora/eglot#709] [Q&A] [answered]
I am trying to migrate my `lsp-mode` setup to eglot, and I am held back by not finding any way to specify settings that are specific to the language server I am using (elixir-ls).  I presently have `(lsp-register-custom-settings '(("elixirLS.mixEnv" "dev")))` in my init, and I cannot figure out what the equivalent would be in eglot.  The docstring for `lsp-register-custom-settings` is as follows:
> (lsp-register-custom-settings PROPS)
> 
> Register PROPS.
> PROPS is list of triple (path value boolean?) where PATH is the path to the
property; VALUE can be a literal value, symbol to be evaluated, or either a
function or lambda function to be called without arguments; BOOLEAN? is an
optional flag that should be non-nil for boolean settings, when it is nil the
property will be ignored if the VALUE is nil.
> 
> Example: ‘(lsp-register-custom-settings ’(("foo.bar.buzz.enabled" t t)))’
(note the double parentheses)

Does eglot provide an equivalent mechanism?

**Comment by funnes** [comment, marked as answer]
I think the variable `eglot-workspace-configuration` is what you are looking for.

I have this in my init to configure `pylsp`:
```elisp
(setq-default eglot-workspace-configuration
 '((pylsp                                  
    (plugins                               
     (jedi_completion (fuzzy . t))        
     (pydocstyle (enabled . t))))))        
```

**Reply by joaotavora** [reply]
This looks fine, but it is only one way to set the `eglot-workspace-configuration` variable.  The README should have more examples.

**Reply by lakkiy** [reply]
```
(setq-default eglot-workspace-configuration
              '((gopls
                 (usePlaceholders . t))))
```

Works for go too.

**Reply by aidalgol** [reply]
So I *finally* got around to trying to switch to eglot again.  This *seems* to be working, but is there any way I can query the workspace configuration to confirm?
```elisp
(setq-default eglot-workspace-configuration
 '((eglot
    (mixenv . "dev"))))
```

**Comment by joaotavora** [comment]
Converted this to a discussion, as general questions are not what issues are for

**Reply by aidalgol** [reply]
Oh, sorry!  I didn't realise GitHub had added a forum feature.
